src/effects/Echo.cpp

```diff
@@ -42,6 +42,43 @@ const ComponentInterfaceSymbol EffectEcho::Symbol
 
 namespace{ BuiltinEffectsModule::Registration< EffectEcho > reg; }
 
+
+struct EffectEcho::Instance
+   : public PerTrackEffect::Instance
+   , public EffectInstanceWithBlockSize
+{
+   Instance(const PerTrackEffect& effect)
+      : PerTrackEffect::Instance{ effect }
+   {}
+
+   void SetSampleRate(double rate) override { mSampleRate = rate; }
```

Comment:
Use EffectInstanceWithSampleRate to simplify.  Remove the member function and variable for mSampleRate.  Just inherit them.

Reply:
ok, done